sdk/clientcore/core/src/main/java/io/clientcore/core/util/ParameterizedTypeImpl.java

```diff
@@ -24,7 +25,12 @@ public final class ParameterizedTypeImpl implements ParameterizedType {
      */
     public ParameterizedTypeImpl(Class<?> raw, Type... args) {
         this.raw = raw;
-        this.args = args;
+
+        Type[] argsCopy = new Type[args.length];
+        for (int i = 0; i < args.length; i++) {
+            argsCopy[i] = args[i];
+        }
+        this.args = argsCopy;
```

Comment:
This might be sufficient (with some null-checking):

```java
this.args = Arrays.copyOf(args, args.length);
```

Reply:
Arrays.copyOf() produces a shallow copy. With shallow copy, the content will still be accessible externally.